### PR TITLE
Add ClientCredentials Authentication with Certificate

### LINKED ISSUE #1093: [BC Idea]: Client Credentials Authentication via Certificate+Password for Microsoft Graph Module + RestClient
### BC Idea Link

https://experience.dynamics.com/ideas/idea/?ideaid=ffb44351-ee10-ef11-9899-000d3ae29eff

### Description

We should be able to use ClientCredentials Flow with Certificate+Password .

### I will provide the implementation for this BC Idea

- [X] I will provide the implementation for this BC Idea
Internal work item: [AB#534910](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/534910)

### PR DESCRIPTION
**Pull request by pri-kise**
#### Summary
Add Authentication with Certificate+Password to RestClient and GraphClient.

#### Work Item(s)
Fixes #1093 

Fixes [AB#534910](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/534910)

### CODE REVIEW

**Review by PeterConijn — APPROVED** [review]
Looks good, but

**Comment by pri-kise** [comment]
@PeterConijn 
> Looks good! I was thinking about automated tests, but since this is all external connection, I'm not sure how you'd easily implement that (aside perhaps injection).

For the existing procedures there are already tests without authentication.

And I've added a similiar type of authentication recently for the SharePoint Module.
https://github.com/microsoft/BCApps/blob/main/src/System%20Application/App/SharePoint%20Authorization/src/SharePointClientCredentials.Codeunit.al
You could check that I didn't made any copy-paste issues and that I didn't forgot any `not` or something similar.

**Comment by JesperSchulz** [comment]
@Darjoo, we would need your sign-off on this one. This PR has security impact. Please sign off on attached slice when completed.

File: src/System Application/App/MicrosoftGraph/src/Authorization/GraphAuthClientCredentials.Codeunit.al [hunk 1]
```diff
@@ -15,25 +15,45 @@ codeunit 9357 "Graph Auth. Client Credentials" implements "Graph Authorization"
...
+        CertificatePassword := NewCertificatePassword;
+        Scopes := NewScopes;
+    end;
+
     procedure GetHttpAuthorization(): Interface "Http Authentication"
     var
         HttpAuthOAuthClientCredentials: Codeunit HttpAuthOAuthClientCredentials;
         OAuthAuthorityUrl: Text;
     begin
+
```
Comment (darjoo) [inline src/System Application/App/MicrosoftGraph/src/Authorization/GraphAuthClientCredentials.Codeunit.al]:
```suggestion
```

File: src/System Application/App/Rest Client/src/Authentication/HttpAuthOAuthClientCredentials.Codeunit.al [hunk 2]
```diff
@@ -75,4 +105,26 @@ codeunit 2361 "HttpAuthOAuthClientCredentials" implements "Http Authentication"
 
         exit(IsSuccess);
     end;
+
+    local procedure AcquireTokenWithCertificate(var AccessToken: SecretText; var ErrorText: Text): Boolean
+    var
+        OAuth2: Codeunit System.Security.Authentication.OAuth2;
+        IsSuccess: Boolean;
+        IdToken: Text;
+        AquireTokenWithCertificateFailedErr: Label 'Acquire of token with Certificate failed.';
```
Comment (darjoo) [inline src/System Application/App/Rest Client/src/Authentication/HttpAuthOAuthClientCredentials.Codeunit.al]:
```suggestion
        AcquireTokenWithCertificateFailedErr: Label 'Acquire of token with Certificate failed.';
```

File: src/System Application/App/Rest Client/src/Authentication/HttpAuthOAuthClientCredentials.Codeunit.al [hunk 3]
```diff
@@ -75,4 +105,26 @@ codeunit 2361 "HttpAuthOAuthClientCredentials" implements "Http Authentication"
...
+        ClearLastError();
+        if (not OAuth2.AcquireTokensFromCacheWithCertificate(ClientIdGlobal, CertificateGlobal, CertificatePasswordGlobal, '', OAuthAuthorityUrlGlobal, ScopesGlobal, AccessToken, IdToken)) or (AccessToken.IsEmpty()) then
+            OAuth2.AcquireTokensWithCertificate(ClientIdGlobal, CertificateGlobal, CertificatePasswordGlobal, '', OAuthAuthorityUrlGlobal, ScopesGlobal, AccessToken, IdToken);
+
+        IsSuccess := not AccessToken.IsEmpty();
+
+        if not IsSuccess then begin
+            ErrorText := GetLastErrorText();
+            if ErrorText = '' then
+                ErrorText := AquireTokenWithCertificateFailedErr;
```
Comment (darjoo) [inline src/System Application/App/Rest Client/src/Authentication/HttpAuthOAuthClientCredentials.Codeunit.al]:
```suggestion
                ErrorText := AcquireTokenWithCertificateFailedErr;
```

**Review by darjoo — APPROVED** [review]

**Review by JesperSchulz — APPROVED** [review]

**Pull request merged**